src/java.base/share/classes/java/security/PEMRecord.java

```diff
@@ -0,0 +1,145 @@
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.security;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+/**
+ * {@code PEMRecord} is a {@link DEREncodable} that represents Privacy-Enhanced
+ * Mail (PEM) data by its type and Base64 form.  {@link PEMDecoder} and
+ * {@link PEMEncoder} use {@code PEMRecord} when representing the data as a
+ * Java API cryptographic object is not desired or there is no other
+ * {@code DEREncodable} for the type.
+ *
+ * <p> Types with Java API representation, such as a {@link PrivateKey},
+ * can return a {@code PEMRecord} when used with
+ * {@linkplain PEMDecoder#decode(String, Class)}. Using {@code PEMRecord} can
+ * be helpful when generating a representation is not desired or when used
+ * with {@code leadingData}.
+ *
+ * <p> {@code PEMRecord} may have a null {@code type} and {@code pem} when
+ * {@code PEMDecoder.decode()} methods encounter only non-PEM data and has
+ * reached the end of the stream. If there is PEM data, {@code type} and
+ * {@code pem} will both be non-null. {@code leadingData} may be null if the
+ * input data only contains PEM data. All values can never be null.
+ *
+ * <p> During the instantiation of this record, there is no validation for the
+ * {@code type} or {@code pem}.
+ *
+ * @param type The type identifier in the PEM header.  For a public key,
+ * {@code type} would be "PUBLIC KEY".
+ * @param pem Any data between the PEM header and footer.
+ * @param leadingData Any non-PEM data read during the decoding process
+ * before the PEM header. This can be useful when reading metadata that
+ * accompanies PEM data.
+ *
+ * @spec https://www.rfc-editor.org/info/rfc7468
+ *       RFC 7468: Textual Encodings of PKIX, PKCS, and CMS Structures
+ */
+public record PEMRecord(String type, String pem, byte[] leadingData)
+    implements DEREncodable {
+
+    /**
+     * Return a PEMRecord instance with the given parameters.
+     *
+     * <p> When {@code type} is given a properly formatted PEM header, only the
+     * identifier will be set (ie: {@code PUBLIC KEY}).  Otherwise, {@code type}
+     * will be set to what was passed in.
```

Comment:
Put PUBLIC KEY in double quotes so it is clear this is a String.

I'm not sure I understand this paragraph - will the dashes be included if the type is not understood? If not, why do we need this paragraph at all?